### PR TITLE
wip: grpc update to 1.25.0

### PR DESCRIPTION
**Pull request by asraa**
WIP: gRPC update to 1.25.0. 
```
ERROR: envoy/bazel/repositories.bzl:692:5: error loading package '@com_github_grpc_grpc//': in /usr/local/google/home/asraa/.cache/bazel/_bazel_asraa/f23f95d44d486dfd95cb67d4d7eb464d/external/com_github_grpc_grpc/bazel/grpc_build_system.bzl: in /usr/local/google/home/asraa/.cache/bazel/_bazel_asraa/f23f95d44d486dfd95cb67d4d7eb464d/external/upb/bazel/upb_proto_library.bzl: Unable to load file '@bazel_version//:bazel_version.bzl': file doesn't exist and referenced by '//external:grpc'
```

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

**Comment by nicolasnoble** [comment]
cc @gnossen

**Comment by asraa** [comment]
I've also tried to patch in this commit https://github.com/grpc/grpc/commit/c22c215ea548838bc4cf2d400fd34ec409cb82af

and load `grpc_extra_deps()`. However, I modified the defn of `grpc_extra_deps()` in the patch to remove `go_rules_dependencies()` and `go_register_toolchains()` since we already load those in the `WORKSPACE`. When I build `//source/exe:envoy-static`, I get
```
 ERROR: error loading package 'source/exe': in /usr/local/google/home/asraa/git/envoy/bazel/envoy_build_system.bzl: in /usr/local/google/home/asraa/.cache/bazel/_bazel_asraa/f23f95d44d486dfd95cb67d4d7eb464d/external/rules_foreign_cc/tools/build_defs/cmake.bzl: in /usr/local/google/home/asraa/.cache/bazel/_bazel_asraa/f23f95d44d486dfd95cb67d4d7eb464d/external/rules_foreign_cc/tools/build_defs/cc_toolchain_util.bzl: Unable to load file '@bazel_version//:def.bzl': file doesn't exist
```

Let me know if you want me to push that change on to this PR!

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -673,16 +673,6 @@ def _fuzzit_linux():
 def _com_github_grpc_grpc():
     _repository_impl(
         "com_github_grpc_grpc",
-        patches = [
```
Comment (htuch) [inline bazel/repositories.bzl]: Should there be a delete of these patches as well in this PR?

Comment (asraa) [reply bazel/repositories.bzl]: Yes, 1.25.0 contains these patches. Sorry this was an early draft just to reproduce the failure I was seeing. Fixed so I don't forget later.

**Comment by gnossen** [comment]
@asraa Pulling in that PR is definitely the way to go. The problem you're seeing afterwards looks like [a name collision between upb and rules_foreign_cc that has already been fixed on their master branch](https://github.com/bazelbuild/rules_foreign_cc/issues/327).

**Comment by asraa** [comment]
Wow, thanks! I updated `rules_foreign_cc` and now I get:
```
ERROR: /usr/local/google/home/asraa/.cache/bazel/_bazel_asraa/f23f95d44d486dfd95cb67d4d7eb464d/external/upb/bazel/workspace_deps.bzl:30:5: no such target '@com_google_protobuf//:third_party/zlib.BUILD': target 'third_party/zlib.BUILD' not declared in package ''; however, a source file of this name exists.  (Perhaps add 'exports_files(["third_party/zlib.BUILD"])' to /BUILD?) defined by /usr/local/google/home/asraa/.cache/bazel/_bazel_asraa/f23f95d44d486dfd95cb67d4d7eb464d/external/com_google_protobuf/BUILD and referenced by '//external:zlib'
```

Looks like something similar to https://github.com/protocolbuffers/protobuf/issues/5472, except in upb?

**Comment by lizan** [comment]
#9041

**Pull request closed without merging**